osu.Framework/Platform/HeadlessGameHost.cs

```diff
@@ -30,10 +30,22 @@ public class HeadlessGameHost : DesktopGameHost
 
         public override IEnumerable<string> UserStoragePaths => new[] { "./headless/" };
 
+        [Obsolete("Use HeadlessGameHost(HostConfig hostConfig) instead.")]
         public HeadlessGameHost(string gameName = null, bool bindIPC = false, bool realtime = true, bool portableInstallation = false)
-            : base(gameName ?? Guid.NewGuid().ToString(), bindIPC, portableInstallation: portableInstallation)
+            : this(new HostConfig
+            {
+                Name = gameName,
+                BindIPC = bindIPC,
+                Realtime = realtime,
+                PortableInstallation = portableInstallation,
+            })
+        {
+        }
+
+        public HeadlessGameHost(HostConfig hostConfig)
+            : base(hostConfig.With(name: hostConfig.Name ?? Guid.NewGuid().ToString()))
```

Comment:
Currently, `HeadlessGameHost` allows parameter-less construction. Same goes for inheritors of `HeadlessGameHost`.

`TestLogging` and some usages osu-side use parameter-less construction, with this, those usages can remain the same.

If this is something we want to keep, we can do this:

```suggestion
        public HeadlessGameHost(HostConfig hostConfig = null)
            : base((hostConfig ??= new HostConfig()).With(name: hostConfig.Name ?? Guid.NewGuid().ToString()))
```

with adequate changes to inheriting classes.

Reply:
I cannot make this because the deprecated constructor still exists and covers the parameterless construction.

Reply:
Actually, if I edit the deprecated constructor to have the first argument mandatory, I could make the new constructor optionally parameterless...

Reply:
Actually, if I edit the deprecated constructor to have the first argument mandatory, I could make the new constructor optionally parameterless...